src/browser/provider/built-in/chrome.js

```diff
@@ -0,0 +1,384 @@
+
+        chromeOpenParameters.cmd = buildChromeArgs(config, cdpPort, chromeOpenParameters.cmd, tempUserDataDir);
+
+        await browserTools.open(chromeOpenParameters, pageUrl);
+
+        await this.waitForConnectionReady(browserId);
+    },
+
+    async openBrowser (browserId, pageUrl, configString) {
+        var config          = await getConfig(configString);
+        var tempUserDataDir = createTempUserDataDir();
+        var cdpPort         = config.cdpPort || await getFreePort();
+
+        await this._startLocalChrome(browserId, config, cdpPort, pageUrl, tempUserDataDir);
+
+        var cdpClientInfo = await getCdpClientInfo(cdpPort, browserId);
+
+        if (cdpClientInfo.client) {
+            await cdpClientInfo.client.Page.enable();
+            await cdpClientInfo.client.Network.enable();
+
+            if (config.emulation)
+                await setEmulation(cdpClientInfo.client, config);
+        }
+
+        Object.assign(cdpClientInfo, { config, cdpPort, tempUserDataDir });
+
+        this.openedBrowsers[browserId] = cdpClientInfo;
+    },
+
+    async closeBrowser (browserId) {
+        var { tab, config, cdpPort } = this.openedBrowsers[browserId];
+
+        if (tab && config.headless)
+            await remoteChrome.closeTab({ id: tab.id, port: cdpPort });
+        else
+            await browserTools.close(browserId);
+
+        if (OS.mac || config.headless)
+            await stopLocalChrome(cdpPort);
+
+        delete this.openedBrowsers[browserId];
+    },
+
+    async isLocalBrowser (browserId, configString) {
+        var config = this.openedBrowsers[browserId] && this.openedBrowsers[browserId].config ||
+                     await getConfig(configString);
+
+        return !config.headless;
+    },
+
+    async takeScreenshot (browserId, path) {
+        var { client, config } = this.openedBrowsers[browserId];
+
+        var screenshot = await client.Page.captureScreenshot({ fromSurface: config.headless });
+
+        await fsWriteFile(path, screenshot.data, { encoding: 'base64' });
+    },
+
+    async resizeWindow (browserId, width, height, currentWidth, currentHeight) {
+        var { client, config, windowId } = this.openedBrowsers[browserId];
+
+        if (config.emulation || !windowId && config.headless) {
+            await setEmulationBounds(client, config.device, width, height);
+
+            return;
+        }
+
+        if (!windowId)
+            return;
+
+        var bounds = await client.Browser.getWindowBounds({ windowId });
+
+        bounds.width += width - currentWidth;
+        bounds.height += height - currentHeight;
+
+        await client.Browser.setWindowBounds({ windowId, bounds });
+    },
+
+    async hasCustomActionForBrowser (browserId) {
+        var { config, windowId, client } = this.openedBrowsers[browserId];
+
+        return {
+            hasResizeWindow:                !!client && (config.emulation || windowId || config.headless),
```

Comment:
is it possible to call `resizeWindow` if run test in emulation mode? For example if we run in the iphone emulator how can we resize the window?